### ISSUE TITLE
LLVM Error on PowerPC64LE: Cannot select ...

### ISSUE THREAD

**Issue by CryZe** [rust-lang/rust#50499]
Since the switch to LLVM 6 my builds on PowerPC64LE have been consistently failing with the following error:
```
LLVM ERROR: Cannot select: 0x7fa027678410: ch = PPCISD::STBRX<ST6[%658](align=1)> 0x7fa02663ce38, 0x7fa0228a63a8, 0x7fa027678ea0, ValueType:ch:i48
  0x7fa0228a63a8: i32 = truncate 0x7fa027938820
    0x7fa027938820: i64 = srl 0x7fa027678750, Constant:i32<16>
      0x7fa027678750: i64 = or 0x7fa0277aebc8, 0x7fa026724680
        0x7fa0277aebc8: i64,ch = CopyFromReg 0x7fa029932440, Register:i64 %200
          0x7fa0279382d8: i64 = Register %200
        0x7fa026724680: i64 = shl nuw nsw 0x7fa027938410, Constant:i32<16>
          0x7fa027938410: i64 = AssertZext 0x7fa027938a28, ValueType:ch:i8
            0x7fa027938a28: i64,ch = CopyFromReg 0x7fa029932440, Register:i64 %202
              0x7fa027938c30: i64 = Register %202
          0x7fa027678958: i32 = Constant<16>
      0x7fa027678958: i32 = Constant<16>
  0x7fa027678ea0: i64,ch = CopyFromReg 0x7fa029932440, Register:i64 %158
    0x7fa0277b0068: i64 = Register %158
In function: _ZN6base646decode13decode_helper17h84289d425683d5b3E.llvm.15558209233772563096
```
[CI Build](https://travis-ci.org/LiveSplit/livesplit-core/jobs/375720556#L754)

This seems to be something that the base64 crate is triggering.

**Comment by nagisa** [comment]
https://bugs.llvm.org/show_bug.cgi?id=32140.

Minimal rust test case:

```rust
#[no_mangle]
pub fn foo(accum: u64, output: &mut [u8;6]) {
    unsafe {
        let bytes = *(&accum.to_be() as *const _ as *const [u8; 8]);
        ::std::ptr::copy_nonoverlapping(bytes.as_ptr(), output.as_mut_ptr(), 6);
    }
}
```

**Comment by cuviper** [comment]
We encountered this in [Fedora bz1554349](https://bugzilla.redhat.com/show_bug.cgi?id=1554349) -- sorry I neglected to report it to Rust too.
We linked it to [LLVM bz 35402](https://bugs.llvm.org/show_bug.cgi?id=35402), now RESOLVED FIXED.

**Comment by cuviper** [comment]
Another PPC64 issue we hit (though big-endian) was [LLVM 37039](https://bugs.llvm.org/show_bug.cgi?id=37039).  I can backport both of these to rust-lang/llvm, if nobody else is already.

**Comment by cuviper** [comment]
Both of those fixes are also in the `release_60` branch destined for the imminent LLVM 6.0.1, so we could also just rebase rust-lang/llvm to pick those up.

**Comment by nikomatsakis** [comment]
cc @cuviper @nagisa -- what is status here, are those backports in progress?

**Comment by cuviper** [comment]
I merged the `release_60` backports from upstream LLVM, and @alexcrichton pulled that into #50514.
